test/index.html

```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE html>
+<html>
+  <head></head>
+  <body>
+    <h1>Hello World!</h1>
+  </body>
+</html>
```

Comment:
@kfrankc I thought the random characters popping up at the end were because we weren't zeroing out the `response_buffer` with `memset` yet. Have you tried adding the newline back in after memsetting?